commcare_connect/form_receiver/processor.py

```diff
@@ -70,15 +70,17 @@ def process_learn_modules(user, xform: XForm, app: CommCareApp, opportunity: Opp
     :param blocks: A list of learn module form blocks."""
     for module_data in blocks:
         module = get_or_create_learn_module(app, module_data)
-        CompletedModule.objects.create(
+        CompletedModule.objects.update_or_create(
```

Comment:
why is this update or create? I do think we should check that the form has not been processed but we should error if we receive two different forms with the same 'module'.

So I think this could be a `get_or_create` using the `xform_id` as one of the query params. If the model exists we verify that it's for the same module and ignore it if it is.

Reply:
I was unsure on what should be done here, so i used the update or create, in case we recieved two entries it would avoid creating duplicate completed modules.
I will update the code to raise an error when it recieves a duplicate completed module. Will do the same for assessments also.

Reply:
Fixed. https://github.com/dimagi/commcare-connect/pull/107/commits/d679bb7c194c8a0ec1a7913f2a16abbff01bae8d